tools/serverpod_cli/lib/src/analyzer/entities/entity_analyzer.dart

```diff
@@ -82,6 +82,8 @@ class SerializableEntityAnalyzer {
       }
     }
 
+    collector.clearErrors();
```

Comment:
Suggestion: Split apart the entity building and error collection.

This looks a little bit lite a Hack to compensate for us running this method twice. Would it be possible to split the concerns a bit more so that we only collect the errors once?

Reply:
A solution could be to only pass down the collector when we want to collect errors, e.g. make it nullable and only pass it along for the second pass?

Reply:
We discussed a cleaner solution today, you can find it in another PR. So instead of running all the validation twice we optimize away the redundant validation and only parse the yaml into our internal representation on the first pass, and only validate a single time on the second pass. This also resulted in some other benefits in solving an issue further up the chain in the language server.